js/test/api.test.ts

```diff
@@ -31,6 +31,97 @@ describe("post", () => {
   });
 });
 
+describe("get", () => {
+  test("resolves with parsed data", async () => {
+    const { promise, resolve } = PromiseWithResolvers<Response>();
+    jest.mocked(fetch).mockReturnValue(promise);
+
+    const result = get({
+      RawData: z.string(),
+      url: "/api/test",
+      parser: (s: string) => s + s,
+    });
+
+    resolve({
+      status: 200,
+      json: () =>
+        new Promise((resolve) => {
+          resolve({ data: "test" });
+        }),
+    } as Response);
+
+    await expect(result).resolves.toBe("testtest");
+  });
+
+  test("rejects on parsing error", async () => {
```

Comment:
**nitpick:** this test name is a big confusing because the error isn't in `parser`, maybe this would be more descriptive?
```suggestion
  test("rejects on return data type error", async () => {
```

Reply:
I think the step that's failing could be called parsing, and zod even calls this `zod.parse()`. I think it's actually `parser` that's misnamed, since that's about transforming typed data into other typed data, rather than turning unknown data into typed data. (The name is left over from before we used zod in Glides, and that step actually did some parsing.)

Reply:
I guess I'd say that really neither should be called parsing strictly speaking because they're not about taking a string and tokenizing it and turning it into an abstract syntax tree, but if Zod uses the terminology then _I guess_ that's okay and it's not a hill I'll choose to die on. 😝